extensions/src/org/pentaho/platform/web/http/api/resources/FileResource.java

```diff
@@ -2228,9 +2228,10 @@ protected ByteArrayOutputStream getByteArrayOutputStream() {
     return new ByteArrayOutputStream();
   }
 
-  protected Document parseText( String text ) throws DocumentException {
+  protected Document parseText( ByteArrayOutputStream outputStream ) throws DocumentException {
```

Comment:
We do not change public API in core classes for no good reason. A protected member can be used by plugins and other code and such changes are only allowed under very limited circumstances (a x.0 release, and the member has been deprecated before so that OEMs can adapt to the change.)

Stick with the string as parameter, then use a StringReader as parameter for the SAXReader. That will do the trick.